python/ray/function_manager.py

```diff
@@ -158,36 +158,37 @@ def export(self, remote_function):
     def fetch_and_register_remote_function(self, key):
         """Import a remote function."""
         (job_id_str, function_id_str, function_name, serialized_function,
-         num_return_vals, module, resources,
-         max_calls) = self._worker.redis_client.hmget(key, [
-             "job_id", "function_id", "function_name", "function",
-             "num_return_vals", "module", "resources", "max_calls"
-         ])
+         module, max_calls) = self._worker.redis_client.hmget(
+             key, [
+                 "job_id", "function_id", "function_name", "function",
+                 "module", "max_calls"
+             ])
         function_id = ray.FunctionID(function_id_str)
         job_id = ray.JobID(job_id_str)
         function_name = decode(function_name)
         max_calls = int(max_calls)
         module = decode(module)
 
-        # This is a placeholder in case the function can't be unpickled. This
-        # will be overwritten if the function is successfully registered.
-        def f(*args, **kwargs):
-            raise RuntimeError("This function was not imported properly.")
-
         # This function is called by ImportThread. This operation needs to be
         # atomic. Otherwise, there is race condition. Another thread may use
         # the temporary function above before the real function is ready.
         with self.lock:
-            self._function_execution_info[job_id][function_id] = (
-                FunctionExecutionInfo(
-                    function=f,
-                    function_name=function_name,
-                    max_calls=max_calls))
             self._num_task_executions[job_id][function_id] = 0
 
             try:
                 function = pickle.loads(serialized_function)
             except Exception:
+
+                def f(*args, **kwargs):
```

Comment:
move this part of code into the exception branch to avoid unnecessary work to do.